uber/custom_tags.py

```diff
@@ -1,55 +1,107 @@
 from uber.common import *
 
 
-@register.filter
-def datetime(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
-    return ' '.join(dt.astimezone(c.EVENT_TIMEZONE).strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm')
+def safe_string(str):
+    return JinjaEnv.env().filters['safe'](str)
 
-from datetime import datetime  # noqa: now that we've registered our filter, re-import the "datetime" class to avoid conflicts
 
+@JinjaEnv.jinja_filter(name='date')
+def date_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
+    # TODO: if this is a datetime.time object, do we have to correct for timezone with .astimezone()?
+    return ' '.join(dt.strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm') if dt else ''
 
-@register.filter
+
+@JinjaEnv.jinja_filter(name='datetime')
+def datetime_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
+    return ' '.join(dt.astimezone(c.EVENT_TIMEZONE).strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm') if dt else ''
+
+
+@JinjaEnv.jinja_filter()
+def yesno(value, arg=None):
+    """
+    PORTED FROM DJANGO
+
+    Given a string mapping values for true, false and (optionally) None,
+    returns one of those strings according to the value:
+
+    ==========  ======================  ==================================
+    Value       Argument                Outputs
+    ==========  ======================  ==================================
+    ``True``    ``"yeah,no,maybe"``     ``yeah``
+    ``False``   ``"yeah,no,maybe"``     ``no``
+    ``None``    ``"yeah,no,maybe"``     ``maybe``
+    ``None``    ``"yeah,no"``           ``"no"`` (converts None to False
+                                        if no mapping for None is given.
+    ==========  ======================  ==================================
+    """
+    if arg is None:
+        arg = 'yes,no,maybe'
+    bits = arg.split(',')
+    if len(bits) < 2:
+        return value  # Invalid arg.
+    try:
+        yes, no, maybe = bits
+    except ValueError:
+        # Unpack list of wrong size (no "maybe" value provided).
+        yes, no, maybe = bits[0], bits[1], bits[1]
+    if value is None:
+        return maybe
+    if value:
+        return yes
+    return no
+
+
+@JinjaEnv.jinja_filter()
 def timestamp(dt):
     from time import mktime
     return str(int(mktime(dt.timetuple())))
 
 
-@register.filter
+@JinjaEnv.jinja_filter()
 def jsonize(x):
-    return SafeString(json.dumps(x, cls=serializer))
+    return safe_string(json.dumps(x, cls=serializer)) if x else '{}'
 
 
-@register.filter
+@JinjaEnv.jinja_filter()
 def subtract(x, y):
     return x - y
 
 
-@register.filter
+@JinjaEnv.jinja_filter()
+def urlencode(s):
+    if type(s) == 'Markup':
```

Comment:
Is this right?  Usually ``type`` returns the actual class which the thing is an instance of:

```python
>>> type('some string')
<class 'str'>
```

I'm sure there's some stuff you can do to get ``type`` to return a string, but I think you might be confusing the Python ``type`` function with the Javascript ``typeof`` operator which DOES return a string.